amazoncommons/amazoncommons.bal

```diff
@@ -0,0 +1,107 @@
+}
+
+# Populate the given headers map with the authorization header. All other headers must be passed into the
+# `orderedHeaders` map in a manner where the keys are ordered; e.g. "X-Amz-Date" must be passed in.
+# + accessKey - The access key
+# + secretKey - The secret key
+# + region - The AWS region, e.g. "us-west-1"
+# + serviceName - The service to be used, e.g. "rekognition"
+# + payload - The payload in text format
+# + canonicalURI - Canonical URI, use "" if nothing
+# + canonicalQueryString - CanonicalQuery String, use "" if nothing
+# + orderedHeaders - The headers to be used in a key ordered manner
+# + method - The HTTP method to be used in API calls
+# + amzdate - The Amzdate
+# + datestamp - The Datestamp
+public function populateAuthorizationHeaders(string accessKey, string secretKey, string region, string serviceName,
+                                             string payload, string canonicalURI, string canonicalQueryString,
+                                             map<string> orderedHeaders, string method, string amzdate, 
+                                             string datestamp) {
+    string canonicalHeaders = "";
+    string signedHeaders = "";   
+    foreach var (k,v) in orderedHeaders {
+        string cnheader = k.toLower();
+        canonicalHeaders = canonicalHeaders + cnheader + ":" + v + "\n";
+        signedHeaders = signedHeaders + cnheader + ";";
+    }    
+    if (signedHeaders.length() > 0) {
+        // remove the extra ";" at the end
+        signedHeaders = signedHeaders.substring(0, signedHeaders.length() - 1);
+    }
+    string payloadHash = encoding:encodeHex(crypto:hashSha256(payload.toByteArray("UTF-8"))).toLower();
+    string canonicalRequest = method + "\n" + canonicalURI + "\n" + canonicalQueryString + "\n" + canonicalHeaders +
+                              "\n" + signedHeaders + "\n" + payloadHash;
+    string credentialScope = datestamp + "/" + region + "/" + serviceName + "/" + "aws4_request";
+    string stringToSign = AWS_SIGN_ALGORITHM + "\n" + amzdate + "\n" + credentialScope + "\n" + 
+                          encoding:encodeHex(crypto:hashSha256(canonicalRequest.toByteArray("UTF-8"))).toLower();
+    byte[] signingKey = getSignatureKey(secretKey, datestamp, region, serviceName);
+    string signature = encoding:encodeHex(crypto:hmacSha256(stringToSign.toByteArray("UTF-8"), signingKey)).toLower();
+    string authorizationHeader = AWS_SIGN_ALGORITHM + " " + "Credential=" + accessKey + "/" + credentialScope + ", " + 
+                                 "SignedHeaders=" + signedHeaders + ", " + "Signature=" + signature;
+
+    orderedHeaders["Authorization"] = authorizationHeader;
+}
+
+function sign(byte[] key, string msg) returns byte[] {
+    return crypto:hmacSha256(msg.toByteArray("UTF-8"), key);
+}
+
+# Generates and returns the AWS signature key
+# + secretKey - the secret key in string format as given in the AWS management console
```

Comment:
Better to add a empty doc line between the description and parameters to increase the readability.